tools/loader/ImageClassifier.cpp

```diff
@@ -475,12 +513,93 @@ int main(int argc, char **argv) {
 
     // Perform the inference execution, updating SMT.
     auto batchSize = inputImageData.dims()[0];
-    loader.runInference(bindings, batchSize);
 
+    loader.runInference(bindings, batchSize);
     // Print the top-k results from the output Softmax tensor.
     numErrors += processAndPrintResults(SMT, inputImageBatchFilenames);
   }
 
+  // Image tensors loaded up to be run at once for benchmark mode.
```

Comment:
`main()` is getting super long at this point. Can we extract all of the newly added logic here could probably go into a helper function?